# http://bugzilla.gnome.org/show_bug.cgi?id=569671
# If pure sniffing yields uncertain results, try again with the filename.
# Patch by Tyler Lawson
--- a/gdk-pixbuf/gdk-pixbuf-io.c	2009-02-09 11:33:45.000000000 -0600
+++ b/gdk-pixbuf/gdk-pixbuf-io.c	2009-02-09 11:36:32.000000000 -0600
@@ -793,8 +793,12 @@
 	gchar **mimes;
 	gchar *type;
 	gint j;
+	gboolean mimetype_uncertain;
 
-	mime_type = g_content_type_guess (NULL, buffer, size, NULL);
+	mime_type = g_content_type_guess (NULL, buffer, size, &mimetype_uncertain);
+	if (mimetype_uncertain)
+		mime_type = g_content_type_guess (filename, buffer, size, NULL);
+		
 
 	for (modules = get_file_formats (); modules; modules = g_slist_next (modules)) {
 		GdkPixbufModule *module = (GdkPixbufModule *)modules->data;
